board/safety/safety_chrysler.h

```diff
@@ -0,0 +1,136 @@
+// board enforces
+//   in-state
+//      ACC is active (green)
+//   out-state
+//      brake pressed
+//      stock LKAS ECU is online
+//      ACC is not active (white or disabled)
+
+// chrysler_: namespacing
+int chrysler_speed = 0;
+
+// silence everything if stock ECUs are still online
+int chrysler_lkas_detected = 0;
+int chrysler_desired_torque_last = 0;
+
+static void chrysler_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
+  int bus_number = (to_push->RDTR >> 4) & 0xFF;
+  uint32_t addr;
+  if (to_push->RIR & 4) {
+    // Extended
+    // Not looked at, but have to be separated
+    // to avoid address collision
+    addr = to_push->RIR >> 3;
+  } else {
+    // Normal
+    addr = to_push->RIR >> 21;
+  }
+
+  if (addr == 0x144 && bus_number == 0) {
+    chrysler_speed = ((to_push->RDLR & 0xFF000000) >> 16) | (to_push->RDHR & 0xFF);
+  }
+
+  // check if stock LKAS ECU is still online
+  if (addr == 0x292 && bus_number == 0) {
+    chrysler_lkas_detected = 1;
+    controls_allowed = 0;
+  }
+
+  // ["ACC_2"]['ACC_STATUS_2'] == 7 for active (green) Adaptive Cruise Control
+  if (addr == 0x1f4 && bus_number == 0) {
+    if (((to_push->RDLR & 0x380000) >> 19) == 7) {
+      controls_allowed = 1;
+    } else {
+      controls_allowed = 0;
+    }
+  }
+
+  // exit controls on brake press by human
+  if (addr == 0x140) {
+    if (to_push->RDLR & 0x4) {
```

Comment:
can we exit on gas too?

Reply:
The stock ACC does not exit on a gas press. The gas pedal is also used to resume ACC when at 0mph for longer than 3 seconds.

If you feel strongly that gas should exit, then I could try to add it. I'd also need to add logic in carstate.py so that OpenPilot doesn't immediately try to turn back on because ACC will still be green:
self.main_on = cp.vl["ACC_2"]['ACC_STATUS_2'] == 7  # ACC is green.
Perhaps add a disabled state that's set when gas is pressed. Then change main_on to be (not disabled) and (acc is green). Disabled would be set back to false when ACC is not green.

Reply:
basically you would need to find a way to disable ACC as well on gas press (that's what we do on Hyundai and Toyota with DSU plugged in). It's fine for now, will be required in order to list this port more officially.

Reply:
Makes sense. I can try to send the ACC cancel button message. It has a counter, so hopefully it doesn't confuse the ECU that typically sends it. I'll add that to my TODO list.